File: Submissions/1014_Best_Sightseeing_Pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& A) {
        int result = 0;
        int prev_spot = A[0];
        for(int i=1; i<A.size(); ++i){
            result = max(result, prev_spot+A[i]-i);
            prev_spot = max(prev_spot, A[i] + i);
        }
        return result;
    }
};

File: Kotlin/1014_Best_Sightseeing_Pair.kt
class Solution {
    fun maxScoreSightseeingPair(values: IntArray): Int {
        var result = 0;
        var prev = values[0];
        
        for(i in 1 until values.size){
            result = Math.max(result, prev + values[i] - i);
            prev = Math.max(prev, values[i] + i);
        }
        
        return result;
    }
}